ooniprobe/Model/Database/Measurement.mm

```diff
@@ -96,4 +96,41 @@ -(void)deleteObject{
     [self remove];
 }
 
+-(void)getExplorerUrl:(void (^)(NSString*))successcb onError:(void (^)(NSError*))errorcb {
+    NSURLComponents *components = [[NSURLComponents alloc] init];
+    components.scheme = @"https";
+    components.host = @"api.ooni.io";
+    components.path = @"/api/v1/measurements";
+    if ([self.test_name isEqualToString:@"web_connectivity"])
+        components.query = [NSString stringWithFormat:@"report_id=%@&input=%@",
+                            self.report_id, self.url_id.url];
+    else
+        components.query = [NSString stringWithFormat:@"report_id=%@",
+                        self.report_id];
```

Comment:
Please, use the NSURLQueryItem API to fill the query items. You can read https://content.pivotal.io/blog/building-urls-with-nsurlqueryitems-and-nsurlcomponents to see how this could be done in practice.